app/uk/gov/hmrc/gform/gform/FormService.scala

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.gform.gform
+import uk.gov.hmrc.gform.models.gform.FormComponentValidation
+import uk.gov.hmrc.gform.validation.FieldOk
+import uk.gov.hmrc.gform.ops.FormComponentOps
+
+class FormService {
+
+  def removeCommas(formValidatedData: List[FormComponentValidation]): List[FormComponentValidation] =
+    formValidatedData.map(removeCommasHelper)
+
+  def removeCommasHelper(formValidationOutcome: FormComponentValidation) = {
+    import formValidationOutcome._
+    if (formComponent.isSterling || formComponent.isPositiveNumber || formComponent.isNumber)
+      FormComponentValidation(
+        formComponent,
+        FieldOk(formComponent, formFieldValidationResult.getCurrentValue.getOrElse("").replaceAll(",", "")))
```

Comment:
Problem with this solution is when input `formValidationOutcome` has no `FieldOk` as its `formFieldValidationResult`. It such a case we possibly discarded error and turn it into `FieldOk` result which is dangerous.

In current implementation this problem is avoided by guarding call to `removeCommas` by check if form is valid. But for the future sake we should implement `removeCommasHelper` in such a way that it modifies `formValidationOutcome` only if it has `FieldOk` as its `formFieldValidationResult`.